src/Data/Hashable/Class.hs

```diff
@@ -776,6 +776,9 @@ instance Hashable1 NE.NonEmpty where
 deriving newtype instance Hashable a => Hashable (Semi.Min a)
 deriving newtype instance Hashable a => Hashable (Semi.Max a)
 
+deriving newtype instance Hashable a => Hashable (Semi.Sum a)
+deriving newtype instance Hashable a => Hashable (Semi.Product a)
```

Comment:
I'd be happy to also add:
- a `@since <version number>` for each new instance
- an accompanying `CHANGELOG` entry

Though what version should be listed? (I notice that some repositories use "unreleased" as a placeholder for the eventual version number.)